internal/app/wwctl/container/exec/main.go

```diff
@@ -20,12 +20,25 @@ import (
 	"github.com/warewulf/warewulf/internal/pkg/wwlog"
 )
 
-/*
-fork off a process with a new PID space
-*/
-func runContainedCmd(args []string) (err error) {
+func runChildCmd(cmd *cobra.Command, args []string) error {
+	child := exec.Command("/proc/self/exe", args...)
+	child.SysProcAttr = &syscall.SysProcAttr{
+		Cloneflags: syscall.CLONE_NEWUTS | syscall.CLONE_NEWPID | syscall.CLONE_NEWNS,
+	}
+	child.Stdin = cmd.InOrStdin()
+	child.Stdout = cmd.OutOrStdout()
+	child.Stderr = cmd.ErrOrStderr()
+	return child.Run()
+}
+
+var childCommandFunc = runChildCmd
+
```

Comment:
For what do we need this variable?

Reply:
It allows the command to be mocked out during testing, since the tests can't actually run `runChildCmd`.

Reply:
See https://github.com/warewulf/warewulf/pull/1251/files#diff-3b25d09560b836b0497ca86e6409117819d182a28a24fc7c182ed06c3006e644R29-R32